### PR TITLE
Bump requirements: ruby 2.5.x and bundler 2.x

### PR DESCRIPTION
**Pull request by jmarrec**
and use pessimistic version constraints for everything (though it doesn"t matter much since they are all `add_development_dependency` and not `add_dependency`)

### CODE REVIEW

**Comment by tijcolem** [comment]
Issue opened to setup CI for Jenkins https://github.com/NREL/cbci_jenkins_libs/issues/79

**Comment by jmarrec** [comment]
I am replaying with a modfied jenkins script that uses the 3.0.0-beta-ruby-slim docker and installs bundler ~> 2.1.

https://ci.commercialbuildings.dev/blue/organizations/jenkins/openstudio-workflow-gem/detail/PR-98/7/pipeline/20

**Comment by nllong** [comment]
@tijcolem can you point this branch to the OpenStudio 3.x Jenkinsfile that works with OpenStudio Workflow? 

@jmarrec Can you review this one last time? I did a bit of cleanup and want to release a PR early 3/23, if possible.

**Comment by nllong** [comment]
@jmarrec specifically can you look at this failing test:

```
bundle exec rspec ./spec/openstudio/workflow/workflow_analysis_spec.rb:849
Run options: include {:locations=>{"./spec/openstudio/workflow/workflow_analysis_spec.rb"=>[849]}}

OSW Integration
  should handle weather file throughout the run (FAILED - 1)

Failures:

  1) OSW Integration should handle weather file throughout the run
     Failure/Error: expect(weather_file_path.empty?).to be false

       expected false
            got true
     # ./spec/openstudio/workflow/workflow_analysis_spec.rb:879:in `block (2 levels) in <top (required)>'

Finished in 0.12411 seconds (files took 0.91857 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/openstudio/workflow/workflow_analysis_spec.rb:849 # OSW Integration should handle weather file throughout the run
```

File: lib/openstudio/workflow/jobs/run_energyplus.rb [hunk 1]
```diff
@@ -52,13 +54,14 @@ def perform
...
     # Checks and configuration
     raise 'No run_dir specified in the registry' unless @registry[:run_dir]
-    ep_path = @options[:energyplus_path] ? @options[:energyplus_path] : nil
+
+    ep_path = @options[:energyplus_path] || nil
     @logger.warn "Using EnergyPlus path specified in options #{ep_path}" if ep_path
 
     @logger.info 'Starting the EnergyPlus simulation'
-    @registry[:time_logger].start('Running EnergyPlus') if @registry[:time_logger]
+    @registry[:time_logger]&.start('Running EnergyPlus')
```
Comment (jmarrec) [inline lib/openstudio/workflow/jobs/run_energyplus.rb]: TIL (today I learned) about the Safe Navigation Operator (&.) in Ruby.

File: lib/openstudio/workflow/version.rb [hunk 2]
```diff
@@ -35,6 +37,6 @@
 
 module OpenStudio
   module Workflow
-    VERSION = '1.3.5'.freeze # Suffixes must have periods (not dashes)
+    VERSION = '2.0.0-pre1' # Suffixes must have periods (not dashes)
```
Comment (jmarrec) [inline lib/openstudio/workflow/version.rb]: This seems to be contradicting the comment

File: spec/files/local_gem_path/gems/color_string-0.0.2/lib/color_string/colorize.rb [hunk 3]
```diff
@@ -1,39 +1,41 @@
+# frozen_string_literal: true
+
 module ColorString
   class Colorize
-    REGEX = /\[[a-z0-9_-]+\]/i
+    REGEX = /\[[a-z0-9_-]+\]/i.freeze
```
Comment (jmarrec) [inline spec/files/local_gem_path/gems/color_string-0.0.2/lib/color_string/colorize.rb]: Isn't the `freeze` redundant here?

File: lib/openstudio/workflow/util/energyplus.rb [hunk 4]
```diff
@@ -345,229 +348,229 @@ def self.merge_output_table_summary_reports(current_object, new_object)
...
-    ValueWhenMaximumOrMinimum,            !- Aggregation Type for Variable or Meter 12
-    Cogeneration:DistrictCooling,!- Variable or Meter 13 Name
-    ValueWhenMaximumOrMinimum;            !- Aggregation Type for Variable or Meter 13
- HEREDOC
+          <<~HEREDOC
+            Output:Table:Monthly,
+              Building Energy Performance - Electricity,  !- Name
+                2,                       !- Digits After Decimal
+                InteriorLights:Electricity,  !- Variable or Meter 1 Name
+                SumOrAverage,            !- Aggregation Type for Variable or Meter 1
```
Comment (jmarrec) [inline lib/openstudio/workflow/util/energyplus.rb]: Not sure I like this indentation change, but I don't like the extra leading spaces will have any effect (that is: end up in the IDF file) so it's probably fine.

Comment (jmarrec) [reply lib/openstudio/workflow/util/energyplus.rb]: Now I get it, you use `<<~HEREDOC` not `<<-HEREDOC` so it ignores the leading whitespace.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,12 @@
 OpenStudio::Workflow Change Log
 ==================================
 
+Version 2.0.0 (Unreleased)
+--------------------------
+
+* Remove support for Ruby 2.2.4
```
Comment (jmarrec) [inline CHANGELOG.md]: + * Add support for Ruby 2.5.5

**Comment by jmarrec** [comment]
I forced pushed after redoing the OSM versiontranslation with the latest develop (3.0.0-rc1).

This test doesn't fail on Ubuntu:

```
$ bundle exec rspec ./spec/openstudio/workflow/workflow_analysis_spec.rb:849

[15:44:53.892950 INFO] Workflow complete
  should handle weather file throughout the run

Finished in 3.77 seconds (files took 0.36983 seconds to load)
1 example, 0 failures
```

File: spec/files/local_gem_path/gems/color_string-0.0.2/lib/color_string/colorize.rb (hunk 3 above)
Comment (nllong) [reply spec/files/local_gem_path/gems/color_string-0.0.2/lib/color_string/colorize.rb]: it is, regex's are not mutable. Rubocop does it anyway ¯\_(ツ)_/¯

File: lib/openstudio/workflow/version.rb (hunk 2 above)
Comment (nllong) [reply lib/openstudio/workflow/version.rb]: ha! valid point. i'll fix this.

File: CHANGELOG.md (hunk 5 above)
Comment (nllong) [reply CHANGELOG.md]: i'll update to support  > 2.5.

File: lib/openstudio/workflow/jobs/run_energyplus.rb (hunk 1 above)
Comment (nllong) [reply lib/openstudio/workflow/jobs/run_energyplus.rb]: my favorite... it was after ruby 2.2 though.

**Comment by nllong** [comment]
ready to merge this down?

**Pull request merged**